web/source/keymanweb.js

```diff
@@ -3206,16 +3191,35 @@ if(!window['keyman']['initialized']) {
       return true;
     }                
 
-    keymanweb.callKeyboardStartGroup = function(Ltarg, Levent) {
+    /**
+     * Function     processKeystroke
+     * Scope        Private
+     * @param       {Object}        device      The device object properties to be utilized for this keystroke.
+     * @param       {Object}        element     The page element receiving input
+     * @param       {Object}        keystroke   The input keystroke (with its properties) to be mapped by the keyboard.
+     * Description  Encapsulates calls to keyboard input processing.
+     * @returns     {number}        0 if no match is made, otherwise 1.
+     */
+    keymanweb.processKeystroke = function(device, element, keystroke) {
       keymanweb._CachedSelectionStart = null; // I3319     
       keymanweb._DeadkeyResetMatched();       // I3318    
       keymanweb.cachedContext.reset();
 
-      // As this function should only be called for hardware keyboard input processing,
-      // this processes the keystroke through hardware-based rules.
-      return keymanweb.processKeystroke(util.physicalDevice, Ltarg, Levent);
+      util.activeDevice = device;
```

Comment:
`activeDevice` doesn't need to be reset after this function, correct?

Reply:
Correct.  It'll always be set by the function call itself.